istioctl/cmd/revision.go

```diff
@@ -0,0 +1,1079 @@
+	tw := new(tabwriter.Writer).Init(w, 0, 0, 1, ' ', 0)
+	tw.Write([]byte("WEBHOOK\tTAG\n"))
+	for _, wh := range desc.Webhooks {
+		tw.Write([]byte(fmt.Sprintf("%s\t%s\n", wh.Name, renderWithDefault(wh.Tag, "<no-tag>"))))
+	}
+	return tw.Flush()
+}
+
+func printControlPlane(w io.Writer, desc *RevisionDescription) error {
+	fmt.Fprintf(w, "\nCONTROL PLANE PODS (ISTIOD): (%d)\n", len(desc.ControlPlanePods))
+	if len(desc.ControlPlanePods) == 0 {
+		if len(desc.Webhooks) > 0 {
+			fmt.Fprintln(w, "No Istiod found in this cluster for the revision. "+
+				"However there are webhooks. It is possible that Istiod is external to this cluster or "+
+				"perhaps it is not uninstalled properly")
+		} else {
+			fmt.Fprintln(w, "No Istiod or the webhook found in this cluster for the revision. Something could be wrong")
+		}
+		return nil
+	}
+	return printPodTable(w, desc.ControlPlanePods)
+}
+
+func printGateways(w io.Writer, desc *RevisionDescription) error {
+	if err := printIngressGateways(w, desc); err != nil {
+		return fmt.Errorf("error while printing ingress-gateway info: %v", err)
+	}
+	if err := printEgressGateways(w, desc); err != nil {
+		return fmt.Errorf("error while printing egress-gateway info: %v", err)
+	}
+	return nil
+}
+
+func printIngressGateways(w io.Writer, desc *RevisionDescription) error {
+	fmt.Fprintf(w, "\nINGRESS GATEWAYS: (%d)\n", len(desc.IngressGatewayPods))
+	if len(desc.IngressGatewayPods) == 0 {
+		if isIngressGatewayEnabled(desc) {
+			fmt.Fprintln(w, "Ingress gateway is enabled for this revision. However there are no such pods. "+
+				"It could be that it is replaced by ingress-gateway from another revision (as it is still upgraded in-place) "+
+				"or it could be some issue with installation")
+		} else {
+			fmt.Fprintln(w, "Ingress gateway is disabled for this revision")
+		}
+		return nil
+	}
+	if !isIngressGatewayEnabled(desc) {
+		fmt.Fprintln(w, "WARNING: Ingress gateway is not enabled for this revision.")
+	}
+	return printPodTable(w, desc.IngressGatewayPods)
+}
+
+func printEgressGateways(w io.Writer, desc *RevisionDescription) error {
+	fmt.Fprintf(w, "\nEGRESS GATEWAYS: (%d)\n", len(desc.IngressGatewayPods))
+	if len(desc.EgressGatewayPods) == 0 {
+		if isEgressGatewayEnabled(desc) {
+			fmt.Fprintln(w, "Egress gateway is enabled for this revision. However there are no such pods. "+
+				"It could be that it is replaced by egress-gateway from another revision (as it is still upgraded in-place) "+
+				"or it could be some issue with installation")
+		} else {
+			fmt.Fprintln(w, "Egress gateway is disabled for this revision")
+		}
+		return nil
+	}
+	if !isEgressGatewayEnabled(desc) {
+		fmt.Fprintln(w, "WARNING: Egress gateway is not enabled for this revision.")
+	}
+	return printPodTable(w, desc.EgressGatewayPods)
+}
+
+type istioGatewayType = string
+
+const (
+	ingress istioGatewayType = "ingress"
+	egress  istioGatewayType = "egress"
+)
+
+func isIngressGatewayEnabled(desc *RevisionDescription) bool {
```

Comment:
it's conventional to exclude "is" from symbol names i.e. ingressGatewayEnabled, here and elsewhere.

Reply:
Fixed it.